File: webmvc/rest-server/src/main/java/com/example/service/UserService.java
package com.example.service;

import java.util.List;

import org.springframework.beans.BeanUtils;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.example.entity.User;
import com.example.repository.UserRepository;

import lombok.RequiredArgsConstructor;

@Service
@Transactional
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public List<User> findAll() {
        return userRepository.findAll();
    }

    public User findById(int id) {
        return userRepository.findById(id).orElseThrow();
    }

    public User create(User user) {
        return userRepository.save(user);
    }

    public User update(int id, User request) {
        User user = userRepository.findById(id).orElseThrow();
        BeanUtils.copyProperties(request, user);
        return userRepository.save(user);
    }

    public void delete(int id) {
        userRepository.deleteById(id);
    }
}

File: others/scala/src/main/scala/com/example/service/UserService.scala
package com.example.service

import com.example.entity.User
import com.example.repository.UserRepository
import javax.persistence.EntityNotFoundException
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

import scala.compat.java8.OptionConverters._
import scala.jdk.CollectionConverters._

@Service
@Transactional
class UserService(val userRepository: UserRepository) {

  def findAll(): Iterable[User] = userRepository.findAll().asScala

  def findByName(name: String): Iterable[User] = userRepository.findByName(name).asScala

  def findById(id: Int): Option[User] = toScala(userRepository.findById(id))

  def getOne(id: Int): User = findById(id).getOrElse(throw new EntityNotFoundException())

  def create(user: User): User = userRepository.save(user)

  def update(user: User): User = {
    val current = getOne(user.getId)
    current.setName(user.getName)
    current.setAge(user.getAge)
    userRepository.save(current)
  }

  def delete(id: Int): Unit = {
    val user = getOne(id)
    userRepository.deleteById(user.getId)
  }
}
